### ISSUE TITLE
Add check for overly sparse system

### ISSUE THREAD

**Issue by JDWarner** [scikit-fuzzy/scikit-fuzzy#82]
@nilayrox18 recently reported an error with this system:

``` python
import numpy as np
import skfuzzy as fuzz

quality = fuzz.Antecedent(np.arange(0,11,1),'quality')
service = fuzz.Antecedent(np.arange(0,11,1),'service')

tip = fuzz.Consequent(np.arange(0,26,1),'tip')

quality.automf(3)
service.automf(3)
tip.automf(3)

quality.view()
service.view()
tip.view()

rule1 = fuzz.Rule(quality['poor'] | service['poor'],tip['poor'])
rule2 = fuzz.Rule(quality['good'] | service['good'],tip['good'])

tip_cnt = fuzz.ControlSystem([rule1, rule2])

tipp = fuzz.ControlSystemSimulation(tip_cnt)

tipp.input['quality'] = 5
tipp.input['service'] = 5

tipp.compute()  # Error here
```

The problem is that for the chosen input values - which are both 5, exactly in the middle of both fuzzy antecedents - neither 'poor' nor 'good' membership functions are active. Thus, there is nothing for the fuzzy system to respond to!

On my system using current master, I get the following traceback: 
``` python
     25 tipp.input['service'] = 5
     26 
---> 27 tipp.compute()

/home/jdwarner/src/scikit-fuzzy/skfuzzy/control/controlsystem.py in compute(self)
    160         for consequent in self.ctrl.consequents:
    161             consequent.output[self] = \
--> 162                 CrispValueCalculator(consequent, self).defuzz()

/home/jdwarner/src/scikit-fuzzy/skfuzzy/control/controlsystem.py in defuzz(self)
    282                              "have at least one rule connected to this "
    283                              "variable and have run the rules calculation.")
--> 284         return defuzz(self.var.universe, output_mf, self.var.defuzzify_method)

/home/jdwarner/src/scikit-fuzzy/skfuzzy/defuzzify/defuzz.pyc in defuzz(x, mfx, mode)
    130     if 'centroid' in mode or 'bisector' in mode:
    131         tot_area = mfx.sum()
--> 132         assert tot_area != 0, 'Total area is zero in defuzzification!'

AssertionError: Total area is zero in defuzzification!
```

There should be a more useful error here, telling the user that for the given inputs all antecedents are returning an activity of zero and the system result cannot be computed. Suggesting their ruleset might be too sparse in the message would give them a much clearer path forward.

We could also return the mean of the consequent(s) in this case, but I'd feel better raising an exception.

**Comment by JDWarner** [comment]
@nilayrox18, if you add a third rule connecting 'average' across the board this system should then work.

**Comment by jsexauer** [comment]
@JDWarner, you are correct, but there is another bug here related to memory views vs lists in python 3 vs python 2.  I've corrected that aspect in a separate PR (#83)

**Comment by JDWarner** [comment]
Closing for now thanks to #97, though a more ideal solution should be considered for the 0.3 dev cycle.
